hw_diag/utilities/hardware.py

```diff
@@ -188,15 +189,35 @@ def detect_ecc(diagnostics):
     # It determines which value will be available for which key
     variant = diagnostics.get('VA')
 
+    i2c_bus = ''
     try:
-        key_storage_bus = get_variant_attribute(variant, 'KEY_STORAGE_BUS')
+        # Example SWARM_KEY_URI: "ecc://i2c-1:96?slot=0"
+        swarm_key_uri = get_variant_attribute(variant, 'SWARM_KEY_URI')
+        pattern = 'i2c-([0-9]+)'
```

Comment:
the idea behind uri was that we will be able to use urlparser. 
eg. parser = urllib.parse("ecc://i2c-1:96?slot=0")
bus_dev = parser.hostname etc for its component parts.

Reply:
yeah, it's elegant way.